### DISCUSSION TITLE
Payment Status on Sales Invoice

### DISCUSSION THREAD

**Discussion by Cattrah** [YetiForceCompany/YetiForceCRM#16221] [How to] [answered]
What are the criteria needed in order for Payment Status to change on the Sales Invoice? I have applied Payments In of status Paid to the Invoice and the Sales Invoice status remains at Not Paid. Or where are the files that govern this automatic logic? My cron jobs are running fine.

![image](https://user-images.githubusercontent.com/101741600/176498148-07d22b61-c06f-4140-8474-883f43e2465d.png)

**Comment by rskrzypczak** [comment, marked as answer]
Looks like everything should be working fine.
The cron job responsible for status updates is: _Batch methods_
So check that it does not report errors.
As for the code, it starts here: https://github.com/YetiForceCompany/YetiForceCRM/blob/developer/modules/PaymentsIn/handlers/PaymentsInHandler.php#L24

**Reply by Cattrah** [reply]
Thanks for the help. It does work in the demo environment and not in mine so I'm trying to find the problem. I don't see any errors in my batch job, is there another place to look other than /cache/logs/cron.log ?

![image](https://user-images.githubusercontent.com/101741600/176747911-3068439a-ff49-4d68-95b5-55863dccbb41.png)

I turned on debug_cron but its not putting any new log files in the cron directory, so maybe it isn't running?
/** Debug cron => cache/logs/cron/ */
public static $DEBUG_CRON = true;

What's odd is that it sometimes will randomly work and a payment status will get updated, but I can't find a rhyme or reason for it. I've also tried just making a workflow to do the same thing and I can't get it to let me update the payment_status field on the Invoice. So maybe the problem isn't that the batch job isn't running but there's something wrong with my payment status field.

The only thing I see different from my environment and the demo is that the Basic Field checkbox exists and is checked in Demo but I don't have that field, is this a setting somewhere? I'm using Yetiforce v 6.3.0
![image](https://user-images.githubusercontent.com/101741600/176750526-1d335331-7b38-43dd-a63e-0b4b65b6393e.png)

![image](https://user-images.githubusercontent.com/101741600/176750819-c8237c89-5bab-4700-b438-f67279dff2e2.png)

**Reply by rskrzypczak** [reply]
The second part of your comment is unrelated.
In general, the status of invoices is updated when a related payment with Paid status is added or the status of the related payment changes. Change the payment status and wait for the cron job to run and see the result.
Since sometimes functionalities work for you, you need to catch the steps for which it doesn't work. There is no such thing as randomness in this case.

**Reply by Cattrah** [reply]
Thanks for clarifying the field settings are unrelated. The batch does seem to be running. Several of the payments/invoices in that our office manager created and set the relationship to are updating the payment status just fine. The puzzling bit is why most of these others are not being updated since I can't find any differences in the payment amounts or statuses that are different from the ones that worked. I know it isn't random, something is wrong with the ones not updating, I just can't see the problem from the front end. Most of the ones that aren't updating are ones I entered with the System Administrator account, but I don't see anything in the code that looks like it matters who entered the invoices or payments. 

![image](https://user-images.githubusercontent.com/101741600/176975033-bc72a27f-6da6-4b86-9a83-b863258bfc73.png)

Also likely unrelated, but what triggers this banner to appear? None of our accounts have it in our installation even though there are clearly outstanding invoices and payments related to the Accounts.

![image](https://user-images.githubusercontent.com/101741600/176975149-2df6a7cc-a3b8-431f-b34e-b50e8e47b81b.png)

Thank you so much for your time and feedback with troubleshooting this issue.

**Reply by rskrzypczak** [reply]
![image](https://user-images.githubusercontent.com/10221999/177083241-5e5fbdbc-8c9f-41a5-acea-e9d2ac777a55.png)
This "banner" includes the _Net_ and _Sale date_ fields from related invoices. If they are not empty, it should appear.

**Comment by Cattrah** [comment]
Thank you for the help! I got too frustrated trying to figure out what was wrong with the batch job, so I used much of the same code to write a custom workflow and now things work as expected. 

Also got our invoices banners working by requiring the sale date. Thanks again! We love Yeti and are constantly being pleasantly surprised by useful and clever features!
